### ISSUE TITLE
Fine-tune result not smooth

### ISSUE THREAD

**Issue by JackHenry1992** [iro-cp/FCRN-DepthPrediction#29]
Hi, @iro-cp , I want to fine-tune from your nyu checkpoint model for my own dataset.

This is my fine-tune loss (berhu as your paper), fine-tune code is similar with [tensorflow-deeplab-resnet](https://github.com/DrSleep/tensorflow-deeplab-resnet/blob/master/fine_tune.py) 
Finetune layer :  layer16x_ and ConvPred
Learning Rate :  0.001
BatchSize : 8
Optimizer : AdamOptimizer

![image](https://user-images.githubusercontent.com/9390228/31324146-3f31bdd6-ace2-11e7-9f12-4090c8de0b18.png)

Tensorboard shows the depth result doesn't smooth in plane scenes (such as wall, red circle)
![image](https://user-images.githubusercontent.com/9390228/31324259-8e991cba-ace3-11e7-8c2f-125334454ff0.png)
![image](https://user-images.githubusercontent.com/9390228/31324525-0f0aba46-ace6-11e7-8698-860be8660b35.png)

Can you provide some advise? 
Thanks very much for sharing this perfect work!

**Comment by JackHenry1992** [comment]
I have solved this issue

**Comment by NikolausDemmel** [comment]
@JackHenry1992, could you please share what the issue and and solution was in the end?

**Comment by JackHenry1992** [comment]
Hi, @NikolausDemmel , the issue is that I can't get smooth depth for plane (such as low texture places), the result seems strange and coarse. So I finetune more layers (such as `res5x` and `layer*`) and decrease learning-rate to 0.0001.

**Comment by NikolausDemmel** [comment]
Thanks for the update.

**Comment by chrirupp** [comment]
We think there are a couple of reason why it could look like this:

- you are not ignoring invalid depth pixels during training
- you do not have enough data / augmentations to learn smooth maps
- your loss seems to be very high. Did you forget to divide by the number of (valid!) pixels?

Cheers,
Christian

**Comment by JackHenry1992** [comment]
Hi @chrirupp , thanks for your advises. 
@chrirupp, as you said, my own dataset maybe too small . 

**Do you have some advises for increasing generalization ability (such as different camera)**

- Will it increase the `generalization ability of fcrn-network`  if I use more dataset to train? 
- Will it be worse if I use different camera to get images?

**Comment by chrirupp** [comment]
Hi @JackHenry1992,

yes, in general more data is better for generalization. 
About different cameras: we found that you will need to correct the predicted depth map by multiplying it with the ratio of focal lengths between new and old camera to get better global scale. 
See our recent CNN-SLAM paper https://arxiv.org/abs/1704.03489 for details on this.

**Comment by Ariel-JUAN** [comment]
hi, I also train the nyu checkpoint model, but I am stuck with making my own dataset. I don't know how to make label. Do you have any ideas. Thanks~

**Comment by Ariel-JUAN** [comment]
Would you share the code for making datasets? Please, thank you!

**Comment by chrirupp** [comment]
The easiest way to create a dataset is to run around with a Kinect or similar sensor.  You can try with simulation but that might be quite some effort to render a synthetic dataset.
We did not make the datasets that we use in the paper. They are publicly available (http://cs.nyu.edu/~silberman/datasets/nyu_depth_v2.html)

**Comment by Ariel-JUAN** [comment]
Thank you~

**Comment by 462525023** [comment]
@JackHenry1992 

Hi.  I am also doing such job, but it did not convergence.
I am so sorry but Can you put your code on GitHub or something else? I want to train it again as reference!
Thank you very much!
